demo-app/src/pages/platform-devices/DevicesMap/DevicesMap.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext } from 'react';
+import { Controls, Map, Marker, MarkerCluster } from 'grommet-leaflet';
+import {
+  Anchor,
+  Box,
+  DataContext,
+  NameValueList,
+  NameValuePair,
+  Text,
+} from 'grommet';
+
+const ClusterDetail = ({ cluster }) => {
+  const childMarkers = cluster.getAllChildMarkers();
+  const locations = {};
+  const deviceTypes = {};
+  const parts = {};
+  childMarkers.forEach(marker => {
+    if (locations[marker.options.detail.properties.location]) {
+      locations[marker.options.detail.properties.location] += 1;
+    } else {
+      locations[marker.options.detail.properties.location] = 1;
+    }
+    if (deviceTypes[marker.options.detail.properties.device_type]) {
+      deviceTypes[marker.options.detail.properties.device_type] += 1;
+    } else {
+      deviceTypes[marker.options.detail.properties.device_type] = 1;
+    }
+    if (parts[marker.options.detail.properties.part]) {
+      parts[marker.options.detail.properties.part] += 1;
+    } else {
+      parts[marker.options.detail.properties.part] = 1;
+    }
+  });
+
+  return (
+    <Box gap="small">
+      <Box gap="xsmall">
+        <Text>{childMarkers.length} devices</Text>
+        <NameValueList nameProps={{ width: 'xsmall' }}>
+          <NameValuePair name="Locations">
+            {Object.entries(locations).length}
+          </NameValuePair>
+          <NameValuePair name="Device types">
+            {Object.entries(deviceTypes).length}
+          </NameValuePair>
+          <NameValuePair name="Parts">
+            {Object.entries(parts).length}
+          </NameValuePair>
+        </NameValueList>
+      </Box>
+      <Anchor alignSelf="end" label="View details" />
+    </Box>
+  );
+};
+
+export const DevicesMap = () => {
+  const mapContainerRef = React.useRef();
+  const dataContext = useContext(DataContext);
+  const { data } = dataContext;
+  const locations = data.map(device => {
```

Comment:
linting comment, can this be `const locations = data.map(device -> device.geometry.coordinates`

Reply:
Yep.

Worth discussing in the future whether the tradeoffs associated with this linting rule and how much value it adds.

Pros - turns three lines into 1, eliminates 7 chars
Cons - pain in the rear while developing. If you want to add log messages for debugging, manipulate values, add conditionals, etc. the excessive adding/removing brackets and return statements is an annoying time suck.

Reply:
Totally fair -- I guess my linter/prettier just automatically reformats those to my suggestion usually. Agree there's utility to not having to adding back in brackets, etc. I'm fine leaving as is.

Reply:
Right, but then you have to undo the automatic if you want to extend the function.

Regardless, done.